flashinfer/jit/cubin_loader.py

```
@@ -117,12 +118,18 @@ def download_file(
 
-                    if attempt < retries:
-                        backoff_delay = delay * (2 ** (attempt - 1))
-                        logger.info(f"Retrying in {backoff_delay} seconds...")
+                    if attempt < retries - 1:
+                        # Equal jitter: uniform[cap, 2*cap] with cap=base*2^(attempt-1).
+                        # Preserves an exponentially-growing lower bound on the delay
+                        # so retries adapt to sustained congestion, while still
+                        # decorrelating the 4 parallel download threads (and many CI
+                        # runners) hitting the same CDN edge.
+                        backoff_cap = delay * (2 ** (attempt - 1))
+                        backoff_delay = backoff_cap + random.uniform(0, backoff_cap)
```

Ruff (0.15.12)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)